sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ModelBridgeInternal.java

```diff
@@ -72,12 +73,12 @@ public static CosmosAsyncDatabaseResponse createCosmosAsyncDatabaseResponse(Reso
         return new CosmosAsyncDatabaseResponse(response, client);
     }
 
-    public static <T> CosmosAsyncItemResponse<T> createCosmosAsyncItemResponse(ResourceResponse<Document> response, Class<T> classType) {
-        return new CosmosAsyncItemResponse<>(response, classType);
+    public static <T> CosmosAsyncItemResponse<T> createCosmosAsyncItemResponse(ResourceResponse<Document> response, Class<T> classType, JsonSerializer jsonSerializer) {
```

Comment:
These APIs are leaking implementation into public API, `Document` is internal only, so this may cause issues with JDK 9+.